From 2cd794a2612abb253f572cb41dce3020e02c4aa6 Mon Sep 17 00:00:00 2001
From: Hawking Zhang <Hawking.Zhang@amd.com>
Date: Thu, 25 Jul 2019 17:22:01 +0800
Subject: [PATCH 3192/4256] drm/amdgpu: correct irq type used for sdma ecc

we should pass irq type, instead of irq client id,
to irq_get/put interface

Signed-off-by: Hawking Zhang <Hawking.Zhang@amd.com>
Reviewed-by: Feifei Xu <Feifei.Xu@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/sdma_v4_0.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/sdma_v4_0.c b/drivers/gpu/drm/amd/amdgpu/sdma_v4_0.c
index 2876cdabef41..1e2f3d949d2e 100644
--- a/drivers/gpu/drm/amd/amdgpu/sdma_v4_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/sdma_v4_0.c
@@ -1704,7 +1704,7 @@ static int sdma_v4_0_late_init(void *handle)
 resume:
 	for (i = 0; i < adev->sdma.num_instances; i++) {
 		r = amdgpu_irq_get(adev, &adev->sdma.ecc_irq,
-			sdma_v4_0_seq_to_irq_id(i));
+				   AMDGPU_SDMA_IRQ_INSTANCE0 + i);
 		if (r)
 			goto irq;
 	}
@@ -1848,7 +1848,7 @@ static int sdma_v4_0_hw_fini(void *handle)
 
 	for (i = 0; i < adev->sdma.num_instances; i++) {
 		amdgpu_irq_put(adev, &adev->sdma.ecc_irq,
-			sdma_v4_0_seq_to_irq_id(i));
+			       AMDGPU_SDMA_IRQ_INSTANCE0 + i);
 	}
 
 	sdma_v4_0_ctx_switch_enable(adev, false);
-- 
2.17.1

